File: src/misc/life-expectancy.ts
/*
 Sources
 http://app.iqpf.org/?locale=en#/guidelines/life-expectancy

 Revised
 2020-12-21
 */

export interface IndividualLifeExpectancy {
    0: number;
    5: number;
    10: number;
    15: number;
    20: number;
    25: number;
    30: number;
    35: number;
    40: number;
    45: number;
    50: number;
    55: number;
    60: number;
    65: number;
    70: number;
    75: number;
    80: number;
    85: number;
    90: number;
    95: number;
    100: number;
}

export interface CombinedLifeExpectancy {
    MALE: IndividualLifeExpectancy;
    FEMALE: IndividualLifeExpectancy;
}

export const LIFE_EXPECTANCY: CombinedLifeExpectancy = {
    MALE: {
        0: 95,
        5: 95,
        10: 95,
        15: 95,
        20: 95,
        25: 95,
        30: 95,
        35: 94,
        40: 94,
        45: 94,
        50: 94,
        55: 94,
        60: 94,
        65: 94,
        70: 94,
        75: 94,
        80: 94,
        85: 95,
        90: 96,
        95: 99,
        100: 102,
    },
    FEMALE: {
        0: 97,
        5: 97,
        10: 97,
        15: 97,
        20: 97,
        25: 97,
        30: 97,
        35: 97,
        40: 97,
        45: 96,
        50: 96,
        55: 96,
        60: 96,
        65: 96,
        70: 96,
        75: 96,
        80: 96,
        85: 97,
        90: 98,
        95: 100,
        100: 103,
    },
};

File: src/commonMain/kotlin/com/equisoft/taxca/misc/LifeExpectancy.kt
/*
Sources
    http://app.iqpf.org/?locale=en#/guidelines/life-expectancy

Revised
    2022-04-30
*/

package com.equisoft.taxca.misc

data class CombinedLifeExpectancy(
    val male: Map<Int, Int>,
    val female: Map<Int, Int>,
)

val LifeExpectancy: CombinedLifeExpectancy = CombinedLifeExpectancy(
    male = mapOf(
        0 to 95,
        5 to 95,
        10 to 95,
        15 to 95,
        20 to 95,
        25 to 95,
        30 to 95,
        35 to 94,
        40 to 94,
        45 to 94,
        50 to 94,
        55 to 94,
        60 to 94,
        65 to 94,
        70 to 94,
        75 to 94,
        80 to 94,
        85 to 95,
        90 to 96,
        95 to 99,
        100 to 102,
    ),
    female = mapOf(
        0 to 97,
        5 to 97,
        10 to 97,
        15 to 97,
        20 to 97,
        25 to 97,
        30 to 97,
        35 to 97,
        40 to 97,
        45 to 96,
        50 to 96,
        55 to 96,
        60 to 96,
        65 to 96,
        70 to 96,
        75 to 96,
        80 to 96,
        85 to 97,
        90 to 98,
        95 to 100,
        100 to 103,
    ),
)
